docs/adv-topics/autopatch.md

```diff
@@ -66,64 +161,279 @@ context = autopatch					; dialing context for phone
+| noct | When this is set to 1, the courtesy tone during an autopatch call will be disabled. The default is to send the courtesy tone whenever the radio user unkeys. |
+| quiet | When set to 1, do not send dial tone, voice responses, just try to connect the call. |
+| voxalways | When set to 1, enables vox mode. |
+| exten | Overrides the default autopatch extension. |
+| nostar | Disables the repeater function prefix |
+
+#### Create a Script to Handle Terminating Calls
+Create the following script to handle terminating calls. Without this script, you will be unable to forcefully terminate **incoming** calls:
+
+```bash
+nano /var/lib/asterisk/hangupPhones
 ```
 
-In **extensions.conf**, a stanza for the autopatch context is added which 
-refers to the peer stanza in iax.conf or sip.conf. 
+Paste this script into the file:
 
-For outbound IAX connections extensions.conf
+```bash
+#!/bin/bash
+# Target pattern to match (adjust as needed for different calls)
+TARGET_PATTERN="IAX2/voipms-"
+# TARGET_PATTERN="PJSIP/flowroute-"
+# Get the list of channels and their concise details
+CHANNELS=$(asterisk -rx "core show channels concise")
+echo "Checking for matching channels..."
+echo "--------------------------------"
+# Iterate over each line to find the target channel
+while IFS= read -r line; do
+    if [[ $line == $TARGET_PATTERN* ]]; then
+        CHANNEL=$(echo "$line" | awk -F'!' '{print $1}')
+        CALLERID=$(echo "$line" | awk -F'!' '{print $3}')
+        echo "Found matching channel: $CHANNEL with Caller ID: $CALLERID"
+        echo "Hanging up channel: $CHANNEL"
+        asterisk -rx "channel request hangup $CHANNEL"
+    fi
+done <<< "$CHANNELS"
 ```
-[autopatch] 
 
-exten => _1NXXNXXXXXX,1,Dial,IAX2/peername/${EXTEN} 
-exten => _1NXXNXXXXXX,2,Congestion 
+Make it executable:
+
+```bash
+chmod +x /var/lib/asterisk/hangupPhones
 ```
-For outbound SIP connections it should look like this:
+
+### extentions.conf
+
+In **/etc/asterisk/extensions.conf**, a stanza for the autopatch context is added which 
+refers to the peer stanza in iax.conf or pjsip.conf.  These examples will likely need to be adjusted to work with a specific provider: 
+
+For outbound IAX connections extensions.conf:
 ```
 [autopatch] 
 
-exten => _1NXXNXXXXXX,1,Dial,SIP/peername/${EXTEN} 
-exten => _1NXXNXXXXXX,2,Congestion 
+; Match any 10-digit number (e.g., North American local format).
+exten => _NXXNXXXXXX,1,Dial(IAX2/voipms/${EXTEN})
+; After the call is completed, hang up the line.
+exten => _NXXNXXXXXX,n,Hangup()
+
+```
+For outbound SIP connections extensions.conf, this example uses IP auth:
+```
+[autopatch]
+; Match any 10-digit number (e.g., North American local format).
+exten => _NXXNXXXXXX,1,Dial(PJSIP/flowroute-endpoint/sip:[tech-prefix]*+1${EXTEN}@us-east-va.sip.flowroute.com)
+; After the call is completed, hang up the line.
+exten => _NXXNXXXXXX,n,Hangup()
+```
+
+Setup 911 extention
```

Comment:
To setup the 911 extension, add :

(and yes, here you also need to correct the spelling of extension :-)